### PR TITLE
Tools: autotest: provide description of exceptions in constructors

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

File: Tools/autotest/common.py [hunk 1]
```diff
@@ -815,9 +812,9 @@ def do_set_mode_via_command_long(self, mode, timeout=30):
                          )
             m = self.mav.recv_match(type='HEARTBEAT',
                                     blocking=True,
-                                    timeout=remaining)
```
Comment (amilcarlucas) [inline Tools/autotest/common.py]: Was this change intentional ?

File: Tools/autotest/common.py [hunk 2]
```diff
@@ -829,14 +826,14 @@ def mavproxy_do_set_mode_via_command_long(self, mode, timeout=30):
         while True:
             remaining = timeout - (self.get_sim_time_cached() - tstart)
             if remaining <= 0:
-                raise AutoTestTimeoutException()
+                raise AutoTestTimeoutException("Failed to change mode")
             self.mavproxy.send("long DO_SET_MODE %u %u\n" %
                                (base_mode, custom_mode))
             m = self.mav.recv_match(type='HEARTBEAT',
                                     blocking=True,
-                                    timeout=remaining)
```
Comment (amilcarlucas) [inline Tools/autotest/common.py]: Was this intentional? shouln't it be done in a separated comit?

File: Tools/autotest/common.py [hunk 3]
```diff
@@ -1263,9 +1252,9 @@ def check_test_syntax(self, test_file):
                 # check for lambda: test_function without paranthesis
                 faulty_strings = re.findall(r"lambda\s*:\s*\w+.\w+\s*\)", f.read())
                 if faulty_strings:
-                    self.progress("Syntax error in autotest lamda at : ")
-                    print(faulty_strings)
-                    raise ErrorException()
+                    desc = ("Syntax error in autotest lamda at : " +
```
Comment (amilcarlucas) [inline Tools/autotest/common.py]: s/lamda/lambda

**Comment by amilcarlucas** [comment]
On the px4-v3 build it throws:
```
/home/runner/ardupilot/libraries/AP_HAL_PX4/Semaphores.cpp: In constructor 'PX4::Semaphore_Recursive::Semaphore_Recursive()':
/home/runner/ardupilot/libraries/AP_HAL_PX4/Semaphores.cpp:23:38: error: 'PTHREAD_MUTEX_RECURSIVE' was not declared in this scope
     pthread_mutexattr_settype(&attr, PTHREAD_MUTEX_RECURSIVE);
                                      ^
compilation terminated due to -Wfatal-errors.
```

File: Tools/autotest/common.py (hunk 1 above)
Comment (peterbarker) [reply Tools/autotest/common.py]: Yes.  (By inspection) it seems you could get a "heartbeat not received" error message when the real problem is that you're out of time.

File: Tools/autotest/common.py (hunk 2 above)
Comment (peterbarker) [reply Tools/autotest/common.py]: Probably should be a separate commit.  *shrugworthy* :-)

File: Tools/autotest/common.py (hunk 3 above)
Comment (peterbarker) [reply Tools/autotest/common.py]: Pre-existing! :-)  And we wouldn't want to mix in unrelated changes ;-)

**Comment by peterbarker** [comment]
@amilcarlucas That's (as far as we can see) just semaphore sucking.
